e2e/cypress/integration/channel/more_public_channels_spec.js

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @channel
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('more public channels', () => {
+    let testUser;
+    let otherUser;
+    let testTeam;
+
+    before(() => {
+        cy.apiInitSetup().then(({team, user}) => {
+            testUser = user;
+            testTeam = team;
+
+            cy.apiCreateUser().then(({user: user1}) => {
+                otherUser = user1;
+                cy.apiAddUserToTeam(testTeam.id, otherUser.id);
+            });
+
+            cy.apiLogin(testUser).then(() => {
+                // # create 30 channels
+                for (let i = 0; i < 30; i++) {
+                    cy.apiCreateChannel(testTeam.id, 'public-channel', 'public-channel');
+                }
+
+                // # Go to town square
+                cy.visit(`/${team.name}/channels/town-square`);
+            });
+        });
+    });
+
+    it('MM-T1664 Channels dont disappear from More Channels modal', () => {
+        // # Login as other user
+        cy.apiLogin(otherUser);
+
+        // # Go to town square
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        // # Go to LHS and click "More..." under Public Channels group
+        cy.get('#publicChannelList').should('be.visible').within(() => {
+            cy.findByText('More...').scrollIntoView().should('be.visible').click();
+        });
+
+        // # Assert that the moreChannelsModel is visible
+        cy.get('#moreChannelsModal').should('be.visible').within(() => {
+            cy.wait(TIMEOUTS.HALF_MIN);
+
+            // # Assert that the moreChannelsList is visible and the number of channels is 31
+            cy.get('#moreChannelsList').should('be.visible').children().should('have.length', 31);
+
+            // # scroll to the middle channel
+            cy.get('#moreChannelsList .more-modal__row').eq(15).scrollIntoView();
+
+            // # scroll to the last channel
+            cy.get('#moreChannelsList .more-modal__row').last().scrollIntoView();
+
+            // # scroll to the first channel
+            cy.get('#moreChannelsList .more-modal__row').first().scrollIntoView();
+
+            // #Assert that the no channel message is not visible
+            cy.get('.no-channel-message').should('not.be.visible');
```

Comment:
I missed to mention during my initial review but let's make this more specific than just asserting for a certain class.

Please create a function, like:
```javascript
function verifyNoChannelToJoinMessage(isVisible) {
    cy.findByText('No more channels to join').should(isVisible ? 'be.visible' : 'not.exist');
    cy.findByText('Click \'Create New Channel\' to make a new one').should(isVisible ? 'be.visible' : 'not.exist');
}
```

Then,
```suggestion
            // # Assert that the "No more channels to join" message does not exist
            verifyNoChannelToJoinMessage(false);
```